lib/plugins/aws/provider/awsProvider.js

```diff
@@ -71,7 +71,14 @@ const impl = {
       if (process.env.AWS_SHARED_CREDENTIALS_FILE) {
         params.filename = process.env.AWS_SHARED_CREDENTIALS_FILE;
       }
+
       const profileCredentials = new AWS.SharedIniFileCredentials(params);
+      if (!(profileCredentials.accessKeyId
+          || profileCredentials.sessionToken
+          || profileCredentials.roleArn)) {
+        throw new Error(`Profile ${profile} does not exist`);
```

Comment:
I went with basic `Error` because `this.serverless.classes.Error` is not available here and importing `ServerlessError` with `require` to be against codebase conventions.

Reply:
By the way, we can think of just throwing a warning instead of an `Error`. @pmuens @simonbuchan ?

Reply:
Note that deploys to the wrong environment can actually be dangerous (information leaks to other clients, etc.), and in any case is likely not what you wanted and can take a while to fix in the named S3 bucket case at least. If it prompted, maybe?

I'm not going to say a warning is completely wrong, but it certainly should have a good reason. Only thing I can think of e.g. if someone can come up with a way this is a (detrimental) breaking change: surely nobody is *depending* on a profile existing? Perhaps if `provider.profile` could explicitly default?

Reply:
Yes, I'd agree with @simonbuchan that we should throw an error in that case. Main reason for that is that the user explicitly decided to deploy with a specific profile and expects Serverless to do exactly that (no matter what).

I've also deployed with a wrong profile a couple of times and it was a frustrating experience.

Reply:
Error 👍

Reply:
also, about Errors, a basic Error would be caught and turned into a Serverless Error anyway. I know we've been using Serverless Error all over the codebase, but I wouldn't say it's a convention per se. Basic Error is shorter and simpler imo.